### PR TITLE
Adding supported authentication methods for the client authenticators

### PR DESCRIPTION
**Pull request by ChinthakaJ98**
Ex: PrivateKeyJWTClientAuthenticator supports private_key_jwt, MutualTLSClientAuthenticator supports tls_client_auth
>
> In addition to it in this PR the versions have been upgraded of the identity.inbound.auth.oauth and org.wso2.carbon.identity.core dependencies as below.
identity.inbound.auth.oauth - from 6.11.119 to 6.11.136
org.wso2.carbon.identity.core - from 5.20.322 to 5.25.383

## Related Issues
https://github.com/wso2/product-is/issues/16921

### CODE REVIEW

File: component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/validator/JWTValidator.java [hunk 1]
```diff
@@ -342,7 +342,7 @@ private OAuthAppDO getOAuthAppDO(String jwtSubject) throws OAuthClientAuthnExcep
         String message = String.format("Error while retrieving OAuth application with provided JWT information with " +
                 "subject '%s' ", jwtSubject);
         try {
-            oAuthAppDO = OAuth2Util.getAppInformationByClientId(jwtSubject);
+            oAuthAppDO = OAuth2Util.getAppInformationByClientIdOnly(jwtSubject);
```
Comment (Thumimku) [inline component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/validator/JWTValidator.java:345]: Why we need this method change?

Comment (ChinthakaJ98) [reply component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/validator/JWTValidator.java:345]: The currently used method is deprecated. Mentioned it in the PR description as well.

**Comment by Thumimku** [comment]
Check the build failure, I think you need ti bump framework version also to get this change
https://github.com/wso2/carbon-identity-framework/blob/e70b7ea6cae27f2fb9a60ba3183558c70e596811/components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityTenantUtil.java#L392

**Comment by Thumimku** [comment]
Check the build failure, I think you need ti bump framework version also to get this change
https://github.com/wso2/carbon-identity-framework/blob/e70b7ea6cae27f2fb9a60ba3183558c70e596811/components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityTenantUtil.java#L392

**Review by janakamarasena — APPROVED** [review]

File: component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/validator/JWTValidator.java (hunk 1 above)
Comment (ChinthakaJ98) [reply component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/validator/JWTValidator.java]: This change was reverted in db5a8f354532fa6943464ade0441b0b1d257e36e. Therefore no method change has been done.

**Pull request merged**